core/src/main/java/io/micronaut/core/annotation/EvaluatedExpressionReference.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2017-2022 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.core.annotation;
+
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Wrapper for annotation value, containing evaluated expressions and
+ * class name for generated expression class. This class is only used
+ * at compilation time as part of compile-time annotation metadata.
```

Comment:
This class needs to be in core module at the moment because `AnnotationValueBuilder#members` method has validity check which has a list of valid types that can be included in annotation metadata. The `EvaluatedExpressionReference` is currently part of compile-time annotation metadata, so it seems like it can't be moved to core-processor module